apis/keda/v1alpha1/triggerauthentication_types.go

```diff
@@ -133,9 +133,17 @@ const (
 // AuthPodIdentity allows users to select the platform native identity
 // mechanism
 type AuthPodIdentity struct {
+	// +kubebuilder:validation:Enum=azure;azure-workload;gcp;aws;aws-eks;aws-kiam
 	Provider PodIdentityProvider `json:"provider"`
 	// +optional
 	IdentityID *string `json:"identityId"`
+	// +optional
+	// RoleArn sets the AWS RoleArn to be used. Mutually exclusive with IdentityOwner
+	RoleArn string `json:"roleArn"`
```

Comment:
I thought about it, but I'm not sure if it's aligned with current implementation. I mean, I see something like:
```yaml
podIdentity:
	aws:
		roleArn: sdfjdsafdsa
```

where provider is a nested struct for each provider instead of a magic string, with the required information. The problem is that IdentityID is already coupled with Azure and I think that this will be a mess before KEDA v3. But It's a good change for v3 (due to required breaking changes) because as you said is more user fiendly

Reply:
Gotcha, thanks for the explanation